nginx_config_reloader/copy_files.py

```diff
@@ -0,0 +1,24 @@
+import logging
+from itertools import chain
+from subprocess import check_output, STDOUT
+
+from nginx_config_reloader.settings import SYNC_IGNORE_FILES
+
+logger = logging.getLogger(__name__)
+
+
+def safe_copy_files(src, dest):
+    cmd = [
+        # Adding a / at the end copies contents of the dir and not the dir itself
+        'rsync', src + '/', dest,
+        # Delete and archive without copying over permissions, aka -da but without -p (-a equals -rlptgoD)
+        '-drltgoD',
+        '--chown', 'root:root',
+        '--copy-links',  # Follow symlinks and copy actual file
+        # Dirs default to 0755 to read. Remove setuid bits. Remove executability for others
+        '--chmod="D755,-s,Fo-wx"',
+    ]
+    cmd.extend(["--exclude=\"{}\"".format(pattern) for pattern in SYNC_IGNORE_FILES])
+    cmd = " ".join(cmd)
+    # shell=True to ensure globs are not escaped
+    check_output(cmd, shell=True, stderr=STDOUT)
```

Comment:
You do not seem to be using the output, maybe make this `check_call` and return (and handle) the retcode?

Reply:
Or does the tool itself spit this out to some log?

Reply:
I use `check_output` because it keeps track of the command output in an easier way. Both have the error message of the subprocess call, but `check_output` has more info in `e.output`:

```
In [12]: try:
    ...:     subprocess.check_output('banaan', shell=True, stderr=subprocess.STDOUT)
    ...: except Exception as e:
    ...:     print("Error: {}".format(e))
    ...:     print("Output: {}".format(e.output))
    ...: 
Error: Command 'banaan' returned non-zero exit status 127.
Output: b'/bin/sh: banaan: command not found\n'

In [13]: try:
    ...:     subprocess.check_call('banaan', shell=True, stderr=subprocess.STDOUT)
    ...: except Exception as e:
    ...:     print("Error: {}".format(e))
    ...:     print("Output: {}".format(e.output))
    ...: 
/bin/sh: banaan: command not found
Error: Command 'banaan' returned non-zero exit status 127.
Output: None
```

`stderr=subprocess.STDOUT` is necessary to capture the output properly, otherwise you get no `e.output`:
```
In [17]: try:
    ...:     subprocess.check_call('banaan', shell=True)
    ...: except Exception as e:
    ...:     print("Error: {}".format(e))
    ...:     print("Output: {}".format(e.output))
    ...: 
/bin/sh: banaan: command not found
Error: Command 'banaan' returned non-zero exit status 127.
Output: None

In [18]: try:
    ...:     subprocess.check_output('banaan', shell=True)
    ...: except Exception as e:
    ...:     print("Error: {}".format(e))
    ...:     print("Output: {}".format(e.output))
    ...: 
/bin/sh: banaan: command not found
Error: Command 'banaan' returned non-zero exit status 127.
Output: b''
```